ui/components/app/toast-master/toast-master-selectors.ts

```diff
@@ -0,0 +1,118 @@
+  if (state.metamask?.surveyLinkLastClickedOrClosed) {
+    return false;
+  }
+
+  const startTime = new Date(`${SURVEY_DATE} ${SURVEY_START_TIME}`).getTime();
+  const endTime = new Date(`${SURVEY_DATE} ${SURVEY_END_TIME}`).getTime();
+  const now = Date.now();
+
+  return now > startTime && now < endTime;
+}
+
+/**
+ * Determines if the privacy policy toast should be shown based on the current date and whether the new privacy policy toast was clicked or closed.
+ *
+ * @param state - The application state containing the privacy policy data.
+ * @returns Boolean is True if the toast should be shown, and the number is the date the toast was last shown.
+ */
+export function getShowPrivacyPolicyToast(state: State): {
+  showPrivacyPolicyToast: boolean;
+  newPrivacyPolicyToastShownDate?: number;
+} {
+  const {
+    newPrivacyPolicyToastClickedOrClosed,
+    newPrivacyPolicyToastShownDate,
+    onboardingDate,
+  } = state.metamask || {};
+  const newPrivacyPolicyDate = new Date(PRIVACY_POLICY_DATE);
+  const currentDate = new Date(Date.now());
+
+  const showPrivacyPolicyToast =
+    !newPrivacyPolicyToastClickedOrClosed &&
+    currentDate >= newPrivacyPolicyDate &&
+    getIsPrivacyToastRecent(newPrivacyPolicyToastShownDate) &&
+    // users who onboarded before the privacy policy date should see the notice
+    // and
+    // old users who don't have onboardingDate set should see the notice
+    (!onboardingDate || onboardingDate < newPrivacyPolicyDate.valueOf());
+
+  return { showPrivacyPolicyToast, newPrivacyPolicyToastShownDate };
+}
+
+/**
+ * Returns true if the privacy policy toast was shown either never, or less than a day ago.
+ *
+ * @param newPrivacyPolicyToastShownDate
+ * @returns true if the privacy policy toast was shown either never, or less than a day ago
+ */
+function getIsPrivacyToastRecent(
+  newPrivacyPolicyToastShownDate?: number,
+): boolean {
+  if (!newPrivacyPolicyToastShownDate) {
+    return true;
+  }
+
+  const currentDate = new Date();
+  const oneDayInMilliseconds = 24 * 60 * 60 * 1000;
+  const newPrivacyPolicyToastShownDateObj = new Date(
+    newPrivacyPolicyToastShownDate,
+  );
+  const toastWasShownLessThanADayAgo =
+    currentDate.valueOf() - newPrivacyPolicyToastShownDateObj.valueOf() <
+    oneDayInMilliseconds;
+
+  return toastWasShownLessThanADayAgo;
+}
+
+export function setNewPrivacyPolicyToastShownDate(time: number) {
+  submitRequestToBackground('setNewPrivacyPolicyToastShownDate', [time]);
+}
+
+export function setNewPrivacyPolicyToastClickedOrClosed() {
+  submitRequestToBackground('setNewPrivacyPolicyToastClickedOrClosed');
+}
+
+export function getNftDetectionEnablementToast(state: State): boolean {
+  return Boolean(state.appState?.showNftDetectionEnablementToast);
+}
+
+export function setShowNftDetectionEnablementToast(
+  value: boolean,
+): PayloadAction<string | ReactFragment | undefined> {
+  return {
+    type: SHOW_NFT_DETECTION_ENABLEMENT_TOAST,
+    payload: value,
+  };
+}
```

Comment:
The three setters here are actions not selectors. It looks like the first two are no longer thunks but the last one is still a redux action. Should all three be in this file?

Reply:
Both would work. I agree with the idea of removing these from the global `actions.ts` and co-locating it with other `toast` things.

Maybe a separate `toast-master-actions.ts` would be just a little bit better for readability.

Reply:
@Gudahtt I thought about it a little more, and actually, are plain `submitRequestToBackground` functions `actions` at all?

Reply:
Definitely would recommend a separate file for these, but no they are not technically Redux actions as we're not dispatching them on a reducer, ultimately just utils in my opinion.